dateutil/parser.py

```diff
@@ -911,47 +910,47 @@ def _parse(self, timestr, dayfirst=None, yearfirst=None, fuzzy=False,
                     # "my time +3 is GMT". If found, we reverse the
                     # logic so that timezone parsing code will get it
                     # right.
-                    if i+1 < len_l and l[i+1] in ('+', '-'):
-                        l[i+1] = ('+', '-')[l[i+1] == '+']
+                    if i + 1 < len_l and l[i + 1] in ('+', '-'):
+                        l[i + 1] = ('+', '-')[l[i + 1] == '+']
```

Comment:
Agreed. The flow if this code is somewhat confusing, so I'm not sure why it needs to be altered, but presumably once we have something that looks like a zone name we'd move this logic into a `tzparser()` that specifically parses time zones anyway.

We'll also need to be able to turn off this logic, because the POSIX spec is incredibly counter-intuitive, so I think a bunch of stuff emits stuff with the exact opposite convention.